docs/DeveloperGuide.md

```diff
@@ -1,13 +1,413 @@
+`Validator` object which validates the command on its formatting and details. Depending on the result of the `Validator`
+, the command will be executed in `BudgetBuddy`. The `Validator` and `Object` genre classes utilizes methods and classes
+present in the [transaction](/src/main/java/seedu/budgetbuddy/transaction) folder.
+
+#### 3.2 Parser Class
+The `Parser` class is to mainly determine whether the user input is valid, and proceed to process the command after.
+It uses Boolean Methods to determine the presence of keywords, and then creates a Validator class to process the command
+should the keywords be present.
+
+The following are some examples:
+
+| Boolean Methods                        | Check if input starts with | Feature Requires | Creates                                            |
+|----------------------------------------|----------------------------|------------------|----------------------------------------------------|
+| ListMonthlyExpensesCommand.isCommand() | list monthly expenses      | input            | ListMonthlyExpensesValidator.processCommand(input) |
+| AddExpenseCommand.isCommand()          | add expense                | input            | AddExpenseValidator.processCommand(input)          | 
+
+The Parser class is also used to parse lines from the `Storage` .txt file and loads it into the main application upon
+restart. This will be covered more under the storage class.
+
+#### 3.3 UI Class
+
+The UI Class is to print out elements of the app in the CLI. It contains many methods used to print general, often-used
+messages such as displayWelcomeMessage() and displayExitMessage().
+
+#### 3.4 Command Class
+
+The `Command` class is a parent class which is responsible for checking for input keywords. It has multiple subclasses,
+which corresponds to a specific function of the application. Each of its subclass checks for specific keywords in the
+user input. If the input is valid as checked by the `isCommand()` method, it will call its corresponding validator.
+
+#### 3.5 Validator Classes
+
+The Validator classes are a group of classes which is responsible for further checks on input, such as whether it has
+the right formatting and contains all necessary details. It will return different subclasses of the command object
+depending on whether the input is valid or not. Once the object is instantiated and returned, the `main` class will call
+`execute` to execute the feature.
+
+##### 3.4.1 and 3.5.1 XML Sequence Diagram
+
+The following sequence diagram shows the process of what happens when a user input is passed through the application,
+until it gets executed.
+
+**Note**: 
+1. Validator class is not a parent class, but we have grouped it under validators to prevent cluttering of 
+the diagrams. 
+2. Validator also returns different type of Command Classes depending on the validity of input, which has been omitted
+for the same reason as above.
+
+![CommandClass.drawio.png](diagrams/CommandClass.drawio.png)
```

Comment:
Image could be clearer / use larger font to ensure readability
![image](https://github.com/user-attachments/assets/a2b33f25-7239-4e55-9dec-08a6187eed8b)